zipline/algorithm.py

```diff
@@ -2216,20 +2227,25 @@ def set_max_order_count(self, max_count):
         max_count : int
             The maximum number of orders that can be placed on any single day.
         """
-        control = MaxOrderCount(max_count)
+        control = MaxOrderCount(max_count, on_error)
         self.register_trading_control(control)
 
     @api_method
-    def set_do_not_order_list(self, restricted_list):
+    def set_do_not_order_list(self, restricted_list, on_error='fail'):
         """Set a restriction on which assets can be ordered.
 
         Parameters
         ----------
         restricted_list : container[Asset]
             The assets that cannot be ordered.
         """
-        control = RestrictedListOrder(restricted_list)
+
+        if isinstance(restricted_list, (list, tuple, set)):
```

Comment:
We might also want something like:

```python
elif not isinstance(restricted_list, Restrictions):
    raise TypeError(...)
```

Alternatively, we could decorate this method with `@expect_types(restricted_list=(Restrictions, tuple, list, set), on_error=str)`.